### PR TITLE
Alison/commissions form

### PR DESCRIPTION
**Pull request by adowski**
## Commissions Form
Form for `Buyer` to request a `Commission` from an `Artist`

### Related PRs
None

### Migrations
Created a `Commissions` controller 

### Tests Performed, Edge Cases
None

### Screenshots
None

CC: @clairelin135

### CODE REVIEW

File: app/assets/javascripts/helpers/api_routes.js [hunk 1]
```diff
@@ -19,5 +19,11 @@ class ApiRoutes {
     }
   }
 
+  get commissions() {
+    return {
+      create  : `/commissions/create/`
```
Comment (clairelin135) [inline app/assets/javascripts/helpers/api_routes.js]: This is a `POST` request. Typically a post request simply consists of the route `model_name/`, rather than `model_name/create`. Let's change this to reflect that!

File: app/controllers/commissions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    @commission.destroy
+  end
+
+  private
+    def set_commission
+      @commission = Commission.find(params[:id])
+    end
+
+    def commission_params
+      params.permit(
```
Comment (clairelin135) [inline app/controllers/commissions_controller.rb]: Nice 👍

File: app/views/commissions/_form.html.erb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+<%= form_with(model: commission, local: true) do |form| %>
```
Comment (clairelin135) [inline app/views/commissions/_form.html.erb]: How is this stuff being generated? A scaffold? We don't need a lot of these files, and our forms belong in the react component. In addition, I don't think that these routes are being routed to via `routes.rb`.

File: app/controllers/commissions_controller.rb [hunk 4]
```diff
@@ -0,0 +1,43 @@
+class CommissionsController < ApplicationController
+  before_action :set_commission, only: [:show, :edit, :update, :destroy]
+
+  def index
```
Comment (clairelin135) [inline app/controllers/commissions_controller.rb]: We don't need a lot of these methods. I think that you may have autogenerated these methods, when in reality, we only need the create, show, update, and destroy methods for now.

File: app/controllers/commissions_controller.rb [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+  def update
+  end
+
+  def destroy
+    @commission.destroy
+  end
+
+  private
+    def set_commission
```
Comment (clairelin135) [inline app/controllers/commissions_controller.rb]: Where are we using this?

File: test/controllers/commissions_controller_test.rb [hunk 6]
```diff
@@ -0,0 +1,48 @@
+require 'test_helper'
+
+class CommissionsControllerTest < ActionDispatch::IntegrationTest
+  setup do
```
Comment (clairelin135) [inline test/controllers/commissions_controller_test.rb]: We also do not need this file

File: test/system/commissions_test.rb [hunk 7]
```diff
@@ -0,0 +1,41 @@
+require "application_system_test_case"
+
+class CommissionsTest < ApplicationSystemTestCase
+  setup do
```
Comment (clairelin135) [inline test/system/commissions_test.rb]: Never seen a file like this before :/ Can you tell me about how this was created?

File: db/seeds.rb [hunk 8]
```diff
@@ -72,3 +72,10 @@
 lisa_request_jobs.buyer = Jobs
 lisa_request_jobs.work = Lisa
 lisa_request_jobs.save!
+
+Gates_commission_Picasso = Commission.create(
+  message: 'Picasso, please paint me a portrait'
```
Comment (clairelin135) [inline db/seeds.rb]: Nice!

File: app/views/commissions/show.html.erb [hunk 9]
```diff
@@ -0,0 +1,4 @@
+<p id="notice"><%= notice %></p>
+
```
Comment (clairelin135) [inline app/views/commissions/show.html.erb]: How were all of these views created? We don't need the bulk of these

File: config/routes.rb [hunk 10]
```diff
@@ -4,9 +4,11 @@
   devise_for :buyers
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
   root 'pages#home'
+  get 'artists/:id' => 'artists#show'
+  post 'commissions/create' => 'commissions#create'
```
Comment (clairelin135) [inline config/routes.rb]: As mentioned before, post request should have the route `post 'commissions/' => 'commissions#create'`. Also, I believe that this controller should exist under the `api` namespace. This means that we want to organize it under the `api/` folder within the controllers because CRUD methods seem more like api methods than independent methods outside of the api folder.

Comment (clairelin135) [reply config/routes.rb]: Check out how the `Works` table was created (and routed to) for a good example.

File: config/routes.rb [hunk 11]
```diff
@@ -7,14 +7,15 @@
 
   get 'artists/:id' => 'artists#show', as: :artistid
   get '/artists/', to: 'artists#all_artists'
-
   get 'buyers/:id' => 'buyers#show', as: :buyerid
+  get 'artists/works/:id' => 'artists#works'
```
Comment (clairelin135) [inline config/routes.rb]: We aren't using this route, so we should delete it

File: config/routes.rb [hunk 12]
```diff
@@ -7,14 +7,15 @@
...
 
   namespace :api, defaults: { format: :json } do
     resources :works, :only => [:show, :create, :update, :destroy]
     resources :artists, :only => [:index, :show, :create, :update, :destroy]
     resources :requests, :only => [:show, :create, :update]
     get 'artists/works/:id' => 'artists#works'
+    post 'commissions/' => 'commissions#create'
```
Comment (clairelin135) [inline config/routes.rb]: we should structure commissions the same way that we structure works, artists, and requests above (i.e. `resources :requests, :only...`

File: app/javascript/components/artist_works.js [hunk 13]
```diff
@@ -30,6 +46,13 @@ class ArtistWorks extends React.Component {
             <p>{work.media}</p>
           </div>
         )}
+
+        <form onSubmit = {this.handleSubmit} name="commissionsForm">
+          <textarea type="TEXT" name="comment" id="comment" value={this.state.comment}
```
Comment (clairelin135) [inline app/javascript/components/artist_works.js]: Separate the `textarea` component by putting each prop on a newline so it is more readable

File: app/javascript/components/artist_works.js [hunk 14]
```diff
@@ -18,6 +21,19 @@ class ArtistWorks extends React.Component {
...
+  handleChange(event) {
+    this.setState({ comment: event.target.value });
+  }
+
+  handleSubmit(event) {
+    const artist_id = this.props.artist.id;
+    const commissions_route = APIRoutes.commissions.create;
+    const buyer_id = this.props.buyer.id;
+
+    Requester.post(commissions_route, { "buyer_id": buyer_id, "artist_id": artist_id, "comment": this.state.comment })
```
Comment (clairelin135) [inline app/javascript/components/artist_works.js]: pull the json object out into a variable `const payload = { "buyer_id": buyer_id"...` with each line containing only 1 key-value pair

File: app/controllers/api/commissions_controller.rb [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+
+  def update
+  end
+
+  def destroy
+    @commission.destroy
+  end
+
+  private
+
```
Comment (clairelin135) [inline app/controllers/api/commissions_controller.rb]: remove newline

File: app/controllers/api/commissions_controller.rb [hunk 16]
```diff
@@ -0,0 +1,28 @@
+class Api::CommissionsController < ApplicationController
+  respond_to :json
+  def show
+
+  end
+
+  def create
+    commission = Commission.new(commission_params)
+  	commission.save!
```
Comment (clairelin135) [inline app/controllers/api/commissions_controller.rb]: remove indent

File: app/assets/javascripts/helpers/api_routes.js [hunk 17]
```diff
@@ -28,5 +28,11 @@ class ApiRoutes {
     }
   }
 
+  get commissions() {
+    return {
+      create  : `/api/commissions`
```
Comment (clairelin135) [inline app/assets/javascripts/helpers/api_routes.js]: the colon should be aligned with all the other colors above

File: app/controllers/artists_controller.rb [hunk 18]
```diff
@@ -5,6 +5,7 @@ def index
 
   def show
     @artist = Artist.find(params[:id])
+    @buyer = current_buyer
```
Comment (clairelin135) [inline app/controllers/artists_controller.rb]: We didn't do this earlier, but at the top of each react component, we should be writing a comment that details the props that are passed into the component. see the BuyerProfile react component for an example

File: app/javascript/components/artist_works.js [hunk 19]
```diff
@@ -6,8 +6,11 @@ class ArtistWorks extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      works: []
+      works: [],
+      comment: "",
```
Comment (clairelin135) [inline app/javascript/components/artist_works.js]: no comma
